src/utils/gh_routing.ts

```diff
@@ -0,0 +1,109 @@
+const ESCAPED_AMPERSAND = "~and~";
```

Comment:
Renamed some methods but otherwise first few methods are copied from https://github.com/allen-cell-animated/website-3d-cell-viewer/blob/main/website/utils/gh_route_utils.ts.